pepdbagent/modules/annotation.py

```diff
@@ -362,6 +405,36 @@ def _add_condition(
 
         return statement
 
+    @staticmethod
+    def _add_date_filter(
+        statement: Select,
+        filter_by: Optional[Literal["submission_date", "last_update_date"]],
+        filter_start_date: Optional[str],
+        filter_end_date: Optional[str] = None,
+    ):
+        """
+        Add filter to where clause to sqlalchemy statement (in project search)
+
+        :param statement: sqlalchemy representation of a SELECT statement with where clause
+        :param filter_by: data to use filter on.
+            Options: ["submission_date", "last_update_date"]
+        :param filter_start_date: Filter start date. Format: "YYYY:MM:DD"
+        :param filter_end_date: Filter end date. Format: "YYYY:MM:DD". if None: present date will be used
+        :return: sqlalchemy representation of a SELECT statement with where clause with added filter
+        """
+        start_date = convert_date_string_to_date(filter_start_date)
+        if filter_end_date:
+            end_date = convert_date_string_to_date(filter_end_date)
+        else:
+            end_date = datetime.now()
+        if filter_by == "submission_date":
+            statement = statement.filter(Projects.submission_date.between(start_date, end_date))
+        elif filter_by == "last_update_date":
+            statement = statement.filter(Projects.last_update_date.between(start_date, end_date))
+        else:
+            raise FilterError("Incorrect filter was provided!")
```

Comment:
```suggestion
            raise FilterError("Invalid filter_by was provided!")
```

Reply:
This is a helpful error. A few more different messages (with same error type) for other types of things the user might do wrong would be nice.